guides/advanced_guides/faceted_search.md

```diff
@@ -48,27 +50,42 @@ $ curl \
 
-1. The facets to filter on with the facet syntax (explained below), thanks to the optional parameter `facetFilters`.
-2. the facets for which to retrieve the matching count for, with the `facets` optional parameter.
+### The facets for which to retrieve the matching count
 
-## Additional fields
+`facets=[<facetName>, <facetName>, ...]`
 
-The result of the search query contains two additional fields if the `facets` parameter has been set:
+This attribute can take two values:
 
-  1. `Facets`: returning a count of candidates for each requested facets in the facets parameter, if set to `[*]` a count for all facets is returned.
-  2. `ExhaustiveFacetsCount` telling whether the above count is exhaustive or approximative.
+- `[<facetName>, <facetName>, ...]` (Optional, array of strings, defaults to `null`)
 
-The syntax for passing arguements through facetFilters is `["facetName:facetValue"] or [["FacetName:FacetValue"]]`
+  Comma-separated list of facets for which to retrieve the matching count. The number of remaining candidates for each specified facet is returned. If a facet name doesn't exist, it will be ignored.
 
-### Example
+- `["*"]`
+
+  The `*` character can also be used. In that case, a count for all facets is returned.
+
+If the `facets` parameter has been set, the returned results will contain two additional fields:
+
+1. `facets`: The number of remaining candidates for each specified facet.
+
+2. `exhaustiveFacetsCount`: Whether the above count is **exhaustive** or **approximative**.
+
+### The facets to filter on
+
+`facetFilters=["facetName:facetValue"]` or `facetFilters=[["facetName:facetValue"]]`
+
+- `["facetName:facetValue"]` (Optional, array of strings)
+
+  An array of strings that contains the facet names and values to filter on.
+
+  - `facetName`: The attribute (the name) of a field used as a facet.
+  - `facetValue`: The value of this facet to filter results on.
 
 `[["color:red", "color:blue"], "kind:t-shirt"] <=> ("color:red" OR "color:blue") AND "kind:t-shirt"`
 inner arrays elements are `OR`ed together, outer array elements are `AND`ed together.
 > Array depth must not be greater than 2, and be at least 1.
 
 Errors are sent to the user on:
 
-  1. Incorrect syntax usage`[["color:red"]]` and `["color:red"]` are examples correct syntax while `"color:red"` is incorrect.
-  2. Faceting on existing facets. It is asked of the user to do faceting on an unregistered or not existing facet. (we could send the list of existing facets to the user).
-
-The facets accepts a `[String]` containing all the facets for which we want to retrieve count information. It defaults to `Null`. If it is set to `["*"]`, counts for all facets set above will be returned. It will return facets based on a best effort to match the user request. If a facet name is not existing it will be ignored.
+1. Incorrect syntax usage`[["color:red"]]` and `["color:red"]` are examples correct syntax while `"color:red"` is incorrect.
```

Comment:
We should probably not list **valid** syntax here because we are talking about invalid ones.

Reply:
This is an outdated version. The page is still WIP so there may be conflicts.